src/util_types/archival_mmr.rs

```diff
@@ -21,6 +21,7 @@ use crate::prelude::twenty_first;
 ///
 /// Merkle Mountain Ranges only know about hashes. When values are to be associated with
 /// MMRs, these values must be stored by the caller, or in a wrapper to this data structure.
+#[derive(Debug, Clone)]
 pub struct ArchivalMmr<Storage: StorageVec<Digest>> {
```

Comment:
I think of this as a structured database. What does it mean to clone a database? I guess it's the connection to the database that's being cloned, which might be unproblematic. Still, not completely clear to me, and I see no good reason to allow cloning here.

Reply:
This is not necessarily a structured database; whether it is depends on the `Storage` type. As I see it, it is perfectly valid to clone an archival Merkle Mountain Range in general.

Also, note the types. `Storage` is a generic type argument. As you surely know, `#[derive(Clone)]` only applies if the generic type implements `Clone`. Here is a [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=4321f807fbf6bafad20c54e8f27c6239) in case you want a refresher.